test/unit/ngsiv2/plugins/bidirectional-plugin_test.js

```diff
@@ -0,0 +1,473 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public
+ * License along with fiware-iotagent-lib.
+ * If not, seehttp://www.gnu.org/licenses/.
+ *
+ * For those usages not covered by the GNU Affero General Public License
+ * please contact with::daniel.moranjimenez@telefonica.com
+ *
+ * Modified by: Daniel Calvo - ATOS Research & Innovation
+ */
+
+'use strict';
+
+var iotAgentLib = require('../../../../lib/fiware-iotagent-lib'),
+    utils = require('../../../tools/utils'),
+    should = require('should'),
+    logger = require('logops'),
+    nock = require('nock'),
+    request = require('request'),
+    moment = require('moment'),
+    contextBrokerMock,
+    iotAgentConfig = {
+        contextBroker: {
+            host: '192.168.1.1',
+            port: '1026',
+            ngsiVersion: 'v2'
+        },
+        server: {
+            port: 4041
+        },
+        types: {
+        },
+        service: 'smartGondor',
+        subservice: 'gardens',
+        providerUrl: 'http://smartGondor.com',
+        deviceRegistrationDuration: 'P1M',
+        throttling: 'PT5S'
+    };
+
+describe('Bidirectional data plugin', function() {
+    var options = {
+        url: 'http://localhost:' + iotAgentConfig.server.port + '/iot/devices',
+        method: 'POST',
+        json:
+            utils.readExampleFile('./test/unit/examples/deviceProvisioningRequests/provisionBidirectionalDevice.json'),
+        headers: {
+            'fiware-service': 'smartGondor',
+            'fiware-servicepath': '/gardens'
+        }
+    };
+
+    beforeEach(function(done) {
+        logger.setLevel('FATAL');
+
+        iotAgentLib.activate(iotAgentConfig, function() {
+            iotAgentLib.clearAll(function() {
+                iotAgentLib.addDeviceProvisionMiddleware(
+                    iotAgentLib.dataPlugins.bidirectionalData.deviceProvision);
+                iotAgentLib.addConfigurationProvisionMiddleware(
+                    iotAgentLib.dataPlugins.bidirectionalData.groupProvision);
+                iotAgentLib.addNotificationMiddleware(
+                    iotAgentLib.dataPlugins.bidirectionalData.notification);
+                done();
+            });
+        });
+    });
+
+    afterEach(function(done) {
+        iotAgentLib.clearAll(function() {
+            iotAgentLib.deactivate(done);
+        });
+    });
+
+    describe('When a new provisioning request arrives to the IoTA with bidirectionality', function() {
+        beforeEach(function() {
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v2/subscriptions', function(body) {
+
+                    // As it can be seen in
```

Comment:
This is a *very* good comment. Maybe is better to place it around L50 (where deviceRegistrationDuration is set) as it applies to all `.post('/v2/subscriptions', ...)` cases in the file and that place is more generic.

Reply:
Fixed in a8cfe2230316e8ad28311423344eed9c925c5b10